python/ccf/merkletree.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.

```

Comment:
Any reason a custom implementation of a Merkle tree vs. e.g. https://github.com/Tierion/pymerkletools was preferred?

Reply:
This implementation was inspired by the above and few other merkle tree implementations. The way CCF does hashing was a bit different from the hashing techniques employed by the other solutions. Also, the added MerkleTree solution has more customized logging and has been simplified given our use-case.